revolv/administrator/views.py

```diff
@@ -29,6 +35,423 @@ def get_context_data(self, **kwargs):
+
+    def clean_project_name(self, name):
+        name = name.replace(' ', '-').replace(';', 'sm').replace(':', 'co').replace("'", 'ap').replace('!', 'ex')
+        return name
+
+    def get_context_data(self, **kwargs):
+        context = super(AdministratorAccountingView, self).get_context_data(**kwargs)
+
+        context['project_names'] = [p.title for p in Project.objects.all()]
+        context['admin_repayments'] = AdminRepayment.objects.repayments()
+        context['admin_reinvestments'] = AdminReinvestment.objects.reinvestments()
+        context['donations'] = Payment.objects.donations()
+        context['jsonurl'] = unicode(str(reverse('administrator:accountingJSON')))
+
+        context['clean_project_names'] = {}
+        clean_project_names = context['clean_project_names']
+        for p in context['project_names']:
+            clean_project_names[self.clean_project_name(p)] = p
+
+        return context
+
+class AccountingJSONView(View):
+    """
+    A view that returns a JsonResponse of all relevant accounting data.
+    This is queried by an AJAX request when creating the Accounting page.
+
+    The data is roughly structured as shown below:
+    data = {
+        'cash_in': {},
+        'cash_out': {},
+        'date_info': {},
+        'project_names': [],
+        'cash_balances': {},
+        'clean_project_names': [],
+        }
+    """
+
+    start_date = datetime.datetime.strptime('2015-01-01', '%Y-%m-%d')
+    end_date = datetime.datetime.now()
+    project_filter = Project.objects.all()
+
+    project_dict = {}
+
+
+    """
+    A helper function that calculates the initial cash balance from the
+    beginning of time to self.start_date
+
+    :return:
+        Returns a number that represents initial cash balance.
+    """
+    def calculateCashBalanceFromStart(self):
+        # THESE ARE DUMMY VARIABLES - USE THE REVOLV EARNINGS MODEL INSTEAD
+        payment_service_fee_rate = 0.01
+        revolv_earnings_rate = 0.07
+        allDonations = sum([payment.amount for payment in Payment.objects.donations(end_date=self.start_date)])
+        allReinvestments = sum([payment.amount for payment in AdminReinvestment.objects.reinvestments(end_date=self.start_date)])
+        allRepayments = sum([payment.amount for payment in AdminRepayment.objects.repayments(end_date=self.start_date)])
+        cashInAdjustments = sum([payment.amount for payment in AdminAdjustment.objects.adjustments(end_date=self.start_date, cash_type='cash_in')])
+        cashOutAdjustments = sum([payment.amount for payment in AdminAdjustment.objects.adjustments(end_date=self.start_date, cash_type='cash_out')])
+
+        cash_in = (allDonations * (1 - payment_service_fee_rate)) + (allRepayments * (1 - revolv_earnings_rate)) + cashInAdjustments
+        cash_out  = cashOutAdjustments + allReinvestments
+        net = cash_in - cash_out
+        return net
+    
+    """
+    A helper function that cleans a project name to make it JQuery 
```

Comment:
What do you mean by "jQuery friendly"? It might just be better to URL encode it ([python urlencode](https://docs.python.org/2/library/urllib.html#urllib.urlencode), [javascript decodeURIComponent](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/decodeURIComponent))